File: src-tauri/src/commands/gateway.rs
use tauri::State;

use crate::application::gateway_service::{self, GatewayStartResponse, GatewayStatusResponse, GatewayStopResponse};
use crate::AppState;

#[tauri::command]
pub async fn gateway_status(state: State<'_, AppState>) -> Result<GatewayStatusResponse, String> {
    gateway_service::status(state.inner()).await.map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn gateway_start(state: State<'_, AppState>) -> Result<GatewayStartResponse, String> {
    gateway_service::start(state.inner()).await.map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn gateway_stop(state: State<'_, AppState>) -> Result<GatewayStopResponse, String> {
    gateway_service::stop(state.inner()).await.map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn gateway_restart(state: State<'_, AppState>) -> Result<GatewayStartResponse, String> {
    gateway_service::restart(state.inner()).await.map_err(|e| e.to_string())
}

File: src/api/gateway.ts
import { invoke } from "@tauri-apps/api/core";
import type { GatewaySettings, GatewayStatus } from "./types";

export interface GatewayStartResponse {
  success: boolean;
  address: string;
}

export interface GatewayStopResponse {
  success: boolean;
  message: string;
}

export const gatewayApi = {
  status: (): Promise<GatewayStatus> => invoke<GatewayStatus>("gateway_status"),

  start: (): Promise<GatewayStartResponse> =>
    invoke<GatewayStartResponse>("gateway_start"),

  stop: (): Promise<GatewayStopResponse> =>
    invoke<GatewayStopResponse>("gateway_stop"),

  restart: (): Promise<GatewayStartResponse> =>
    invoke<GatewayStartResponse>("gateway_restart"),

  getSettings: (): Promise<GatewaySettings> =>
    invoke<GatewaySettings>("get_gateway_settings"),

  updateSettings: (data: Partial<GatewaySettings>): Promise<GatewaySettings> =>
    invoke<GatewaySettings>("update_gateway_settings", { payload: data }),
};
